test/e2e/operator_test.go

```diff
@@ -152,6 +159,85 @@ func TestOperatorAvailable(t *testing.T) {
 	}
 }
 
+func TestExternalDNSWithRoute(t *testing.T) {
+	t.Log("Ensuring test namespace")
+	err := kubeClient.Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to ensure namespace %s: %v", testNamespace, err)
+	}
+
+	t.Log("Creating external dns instance with source type route")
+	extDNS := helper.buildOpenShiftExternalDNS(testExtDNSName, hostedZoneID, hostedZoneDomain)
+	if err := kubeClient.Create(context.TODO(), &extDNS); err != nil {
+		t.Fatalf("Failed to create external DNS %q: %v", testExtDNSName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), &extDNS)
+
+	// create a route with the annotation targeted by the ExternalDNS resource
+	t.Log("Creating source route")
+	testRouteHost := "myroute." + hostedZoneDomain
+	route := testRoute(testRouteName, testNamespace, testRouteHost, testServiceName)
+	if err := kubeClient.Create(context.Background(), route); err != nil {
+		t.Fatalf("Failed to create test route %s/%s: %v", testNamespace, testRouteName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), route)
+	t.Logf("Created Route Host is %v", testRouteHost)
+
+	// get the router canonical name
+	var targetRoute routev1.Route
+	if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+		t.Log("Getting canonical host name of create route")
+		err = kubeClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      testRouteName,
+		}, &targetRoute)
+		if err != nil {
+			return false, err
+		}
+
+		// if the status ingress slice is not populated by the ingress controller, try later
+		if len(targetRoute.Status.Ingress) < 1 {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Failed to retrieve the created route %s/%s: %v", testNamespace, testRouteName, err)
+	}
+
+	t.Logf("Target route ingress is %v", targetRoute.Status.Ingress)
+
+	targetRouterCName := targetRoute.Status.Ingress[0].RouterCanonicalHostname
+	if targetRouterCName == "" {
+		t.Fatalf("Created route's canonical name is empty %v", err)
+	}
+	t.Logf("Target route CName is %v", targetRouterCName)
+
+	// try all nameservers and fail only if all failed
+	for _, nameSrv := range nameServers {
+		t.Logf("Looking for DNS record in nameserver: %s", nameSrv)
+
+		// verify dns records has been created for the route host.
+		if err := wait.PollImmediate(dnsPollingInterval, dnsPollingTimeout, func() (done bool, err error) {
+			cnames, err := lookupCNAMEMiekg(testRouteHost, nameSrv)
+			if err != nil {
+				t.Logf("could not find cname for route host %v within name server %v: %v", testRouteHost, nameSrv, err)
```

Comment:
```suggestion
				t.Logf("Waiting for DNS record: %s, error: %v", testRouteHost, err)
```
Just to be consistent on the log messages with the other test.